rust-guest/src/context/context.rs

```diff
@@ -0,0 +1,102 @@
+#![allow(dead_code)]
+#![allow(unused_imports)]
+#![allow(unused_variables)]
+
+use lazy_static::lazy_static;
+use std::sync::Mutex;
+
+use std::io::Cursor;
+use std::mem;
+use std::collections::HashMap;
+use super::generated::{Encode, Decode, Context, Request, Response};
+
+lazy_static! {
+    pub static ref PTR_LEN: Mutex<(u32, u32)> = Mutex::new((0, 0));
```

Comment:
We should probably have these be two separate variables if we're going to do it this way - otherwise in an ideal world we'd just be able to allocate a polyglot buffer but I guess that's not that easy to do.

Reply:
I'm going to revisit this on Monday actually. I realized may have implemented a different global pattern that caused problems with wrapping the buffer when I was first experimenting. Will update here.

Reply:
I was able to get this to a happy-medium. We will still need to save PTR/LEN (now separate) and we have global buffer to help reuse the allocation after the module executes. The problem is with the the global implementation (via lazy_static) requires a Mutex which implements os level resource protections, so we still need to reconstruct the Vec after `resize` to actually populate the buffer after the host calls `write`. See [these](https://github.com/loopholelabs/scale/pull/16/commits/4d5972112389a7e6b59b7a3d211c7373b767d08d) [two](https://github.com/loopholelabs/scale/pull/16/commits/8f10be47688c076f16fead9ce0e9d3bfa36ffc9e) commits for more context.